airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalUtils.java

```diff
@@ -90,15 +96,25 @@ public static WorkflowServiceStubs createTemporalService() {
   private static WorkflowServiceStubsOptions getCloudTemporalOptions() {
     final InputStream clientCert = new ByteArrayInputStream(configs.getTemporalCloudClientCert().getBytes(StandardCharsets.UTF_8));
     final InputStream clientKey = new ByteArrayInputStream(configs.getTemporalCloudClientKey().getBytes(StandardCharsets.UTF_8));
+    WorkflowServiceStubsOptions.Builder optionBuilder;
     try {
-      return WorkflowServiceStubsOptions.newBuilder()
+      optionBuilder = WorkflowServiceStubsOptions.newBuilder()
           .setSslContext(SimpleSslContextBuilder.forPKCS8(clientCert, clientKey).build())
-          .setTarget(configs.getTemporalCloudHost())
-          .build();
+          .setTarget(configs.getTemporalCloudHost());
     } catch (final SSLException e) {
       log.error("SSL Exception occurred attempting to establish Temporal Cloud options.");
       throw new RuntimeException(e);
     }
+
+    MeterRegistry registry = MetricClientFactory.getMeterRegistry();
+    if (registry != null) {
+      StatsReporter reporter = new MicrometerClientStatsReporter(registry);
+      Scope scope = new RootScopeBuilder()
+          .reporter(reporter)
+          .reportEvery(com.uber.m3.util.Duration.ofSeconds(10));
+      optionBuilder.setMetricsScope(scope);
+    }
+    return optionBuilder.build();
```

Comment:
for code readability, i think this should be moved into a private method called `configureTemporalMetricEmission`. In this way, any reader will immediately know what this is for. What do you think?